backend/src/controllers/user.ts

```diff
@@ -51,3 +58,34 @@ export const createUser = async (
 
   return;
 };
+
+export const loginUser = async (
+  req: Request<Record<string, never>, Record<string, never>, LoginUserRequestBody>,
+  res: Response,
+  nxt: NextFunction,
+) => {
+  try {
+    // Check for validation errors
+    const errors = validationResult(req);
+    validationErrorParser(errors);
+
+    const { email, password } = req.body;
+
+    // Create user in Firebase
+    await signInWithEmailAndPassword(firebaseAuth, email, password)
+      .then((userCredential) => {
+        const user = userCredential.user;
+        console.log(user);
+        res.status(200).json(user.uid);
```

Comment:
I would suggest also sending account status (approved or waiting for approval) since after logging in, the page they get directed to is based on whether they've been approved or not.

Reply:
Done